### PR TITLE
Feature/frontend/create new channel/karanmeetkhatra10

### PR DESCRIPTION
**Pull request by karanmeetkhatra10**

### CODE REVIEW

File: Online-Messaging-App/src/app/app.module.ts [hunk 1]
```diff
@@ -22,7 +23,9 @@ import {RegisterModule} from "./register/register.module";
         RegisterModule
     ],
 
-    bootstrap: [AppComponent]
+    bootstrap: [AppComponent],
+    entryComponents: [CreateChannelComponent]
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/app.module.ts]: No, this should be in Home module

File: Online-Messaging-App/src/app/home/channel-browser/channel-browser.component.ts [hunk 2]
```diff
@@ -24,18 +30,32 @@ const DEFAULT_CHANNEL_ROLE: string = "user";
...
 
     constructor(private http: HttpClient, private auth: AuthenticationService) {
     }
 
+    get newChannel(): ChannelObject {
+        return this._newChannel;
+    }
+
+    @Input()
+    set newChannel(value: ChannelObject) {
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/channel-browser/channel-browser.component.ts:50]: add return type

File: Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts [hunk 3]
```diff
@@ -124,4 +133,25 @@ export class SidebarComponent implements OnInit {
...
+    joinChannel():void{
+        const dialogConfig = new MatDialogConfig();
+        dialogConfig.disableClose = true;
+        dialogConfig.autoFocus = true;
+        dialogConfig.width = "35%";
+        let dialogRef = this.dialog.open(CreateChannelComponent,dialogConfig);
+        dialogRef.afterClosed().subscribe((result) => {
+            if(result){
+                this.newChannelEvent.emit(result);
+                this.userSubscribedChannels.push(result);
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts:151]: pull this and line 63 into a common method, and call them in both places

**Review by dBaiko — CHANGES_REQUESTED** [review]
Looks good, just a few minor things

File: Online-Messaging-App/src/app/home/sidebar/sidebar.component.scss [hunk 4]
```diff
@@ -52,6 +52,9 @@ mat-sidenav-container {
     background-color: #282828;
     height: 100%;
 }
+.scrollable{
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/sidebar/sidebar.component.scss]: image
Is there not a way to only make this appear when the list actually needs to be scrolled?

Comment (dBaiko) [reply Online-Messaging-App/src/app/home/sidebar/sidebar.component.scss]: like the scroll bar I mean

File: Online-Messaging-App/src/app/home/createChannel/create-channel.component.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
+.center{
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/createChannel/create-channel.component.scss]: image
How this looks on a bigger monitor btw. If possible please fix

File: Online-Messaging-App/src/app/app.module.ts (hunk 1 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/app.module.ts]: DOne

File: Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts (hunk 3 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts:151]: done

File: Online-Messaging-App/src/app/home/sidebar/sidebar.component.scss (hunk 4 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/sidebar/sidebar.component.scss]: Changed up the code but not sure if it works as I can't recreate issue on my end

File: Online-Messaging-App/src/app/home/createChannel/create-channel.component.scss (hunk 5 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/createChannel/create-channel.component.scss]: Same issue as above, changed some code up but can't recreate issue

File: Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts (hunk 3 above)
Comment (dBaiko) [reply Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts:151]: I meant like line 147 - line 154. That whole thing is reapeated twice, and can be ulled into a method

**Review by dBaiko — APPROVED** [review]

**Pull request merged**